src/eeeoh/eeeoh.ts

```diff
@@ -53,23 +43,28 @@ export type DatadogConfig<CustomLevels extends string = never> =
 
 type LevelToTier = (level: number) => DatadogTier | false | null;
 
-const parseTierByLevelMap = dictionary<string, DatadogTier>(
-  parseString,
-  parseDatadogTier,
-);
+const tierByLevelMapSchema: S.Schema<Partial<Record<string, DatadogTier>>> =
```

Comment:
By default `S.record(datadogTierSchema)` produces `Record<string, DatadogTier>` but our types require: Partial<Record<string, DatadogTier>> which is pretty much the same thing.